ocaml/xapi/xapi_host.ml

```diff
@@ -1400,10 +1399,10 @@ let install_server_certificate ~__context ~host ~certificate ~private_key
   Xapi_mgmt_iface.reconfigure_stunnel ~__context
 
 let emergency_reset_server_certificate ~__context =
-  let args =
-    [!Xapi_globs.server_cert_path; Helper_hostname.get_hostname (); "--force"]
-  in
-  ignore @@ Gencert.call_generate_ssl_cert ~args ;
+  let xapi_ssl_pem = !Xapi_globs.server_cert_path in
+  let common_name = Helper_hostname.get_hostname () in
+  let alt_names = [] in
```

Comment:
I think  Gencert.get_hostnames would better as it would give us alt names as well.

I would also be tempted to replace the implementation of get_hostname with the get_hostnames, but I think that's out of scope of this PR

Reply:
That function is not in the libray but in the binary - so would have to move it.